.github/workflows/release.yaml

```diff
@@ -64,74 +66,119 @@ jobs:
+          echo "//registry.npmjs.org/:_authToken=${{ secrets.NPM_PUBLISH_TOKEN }}" >> .npmrc
+        env:
+          GH_LERNA_TOKEN:    ${{ secrets.GH_LERNA_TOKEN }}
+          NPM_PUBLISH_TOKEN: ${{ secrets.NPM_PUBLISH_TOKEN }}
+
+      # ────────────────────────────────────────────────────────
+      # 1️⃣  Master: version bump via PR (no direct push)
+      # ────────────────────────────────────────────────────────
+      - name: Generate version bump (no-push)
+        if: github.ref == 'refs/heads/master'
+        run: |
+          yarn lerna version \
+            --conventional-commits \
+            --yes \
+            --force-publish \
+            --no-push
+
+      - name: Create bump PR
+        if: github.ref == 'refs/heads/master'
+        uses: peter-evans/create-pull-request@v5
+        with:
+          token: ${{ secrets.GH_LERNA_TOKEN }}
+          commit-message: "chore(release): bump versions [skip ci]"
+          branch: "release/bumps/${{ github.sha }}"
+          title: "chore(release): bump versions"
+          body: |
+            This PR was auto-generated to bump package versions.
+          labels: auto-release
+
+      # ────────────────────────────────────────────────────────
+      # 2️⃣  Master: remove RC tags via PR after RC-branch merge
+      # ────────────────────────────────────────────────────────
+      - name: Stage RC-tag removal
+        if: >-
+          github.ref == 'refs/heads/master' &&
+          contains(github.event.head_commit.message, 'Merge pull request') &&
+          contains(github.event.head_commit.message, 'release-') &&
+          contains(github.event.head_commit.message, '[release]')
         run: |
+            node scripts/remove-rc-tags.js
+            node scripts/remove-rc-tags.js
+            git config --global user.email "cognite-cicd@users.noreply.github.com"
+            git config --global user.name "Cognite CICD"
             node scripts/remove-rc-tags.js
             git config --global user.email "cognite-cicd@users.noreply.github.com"
             git config --global user.name "Cognite CICD"
             git add packages/*/package.json
-            git commit -m "chore: remove rc tags [skip-ci]"
-            git push origin master 
 
-      - name: Lerna version
-        env:
-          GH_TOKEN: ${{ secrets.GH_LERNA_TOKEN }} # Temporary fix to make lerna able to push the new versions commit to master
-          GITHUB_TOKEN: ${{ secrets.GH_LERNA_TOKEN }} # Temporary fix to make lerna able to push the new versions commit to master
+      - name: Create remove-rc PR
+        id: create-remove-rc-pr
+        if: >-
+          github.ref == 'refs/heads/master' &&
+          contains(github.event.head_commit.message, 'Merge pull request') &&
+          contains(github.event.head_commit.message, 'release-') &&
+          contains(github.event.head_commit.message, '[release]')
+        uses: peter-evans/create-pull-request@v5
```

Comment:
not sure about this part at all, we need someone who wrote the RC tag removal logic in a first place. maybe we don't need it anymore?